### PR TITLE
Tool recommendation tool

### PR DESCRIPTION
**Pull request by anuprulez**
- Update versions of dependencies
- Add more tests
- Make changes to `validation_data` attribute for `Keras=2.3.0`
- Add feature to stop training when loss stops dropping (`early stopping`)
- Remove unused parameters

### CODE REVIEW

**Comment by nsoranzo** [comment]
@anuprulez Can you please rebase the PR?

File: tools/tool_recommendation_model/create_tool_recommendation_model.xml [hunk 1]
```diff
@@ -2,11 +2,11 @@
     <description>using deep learning</description>
     <requirements>
         <requirement type="package" version="3.6">python</requirement>
-        <requirement type="package" version="1.14.0">tensorflow</requirement>
```
Comment (bgruening) [inline tools/tool_recommendation_model/create_tool_recommendation_model.xml]: is 1.14 broken?

**Comment by anuprulez** [comment]
I am not sure why the tests are not running

**Comment by bgruening** [comment]
They are running too long with no output, its a Travis limitation which we need to work around ... :(

**Comment by anuprulez** [comment]
@bgruening I reduced the number of iterations in the tests for them to pass. The first two tests take 2-3 minutes to finish in my system. Let's see if they pass now :)

**Comment by anuprulez** [comment]
@bgruening the tests passed :)

**Pull request merged**